### PR TITLE
Ensure the Docker nginx user can write servers.json.

### LINKED ISSUE #1046: 4.0.0 with env based setup fails to start
### shlink-web-client version

4.0.0

### How do you use shlink-web-client

Docker image

### Current behavior

```
docker run --rm -it -e SHLINK_SERVER_URL="https://go.example.com" -e SHLINK_SERVER_API_KEY=foo shlinkio/shlink-web-client:4.0.0
/docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
/docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
/docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
10-listen-on-ipv6-by-default.sh: info: can not modify /etc/nginx/conf.d/default.conf (read-only file system?)
/docker-entrypoint.sh: Sourcing /docker-entrypoint.d/15-local-resolvers.envsh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/30-shlink-servers-json.sh
/docker-entrypoint.d/30-shlink-servers-json.sh: line 11: can't create /usr/share/nginx/html/servers.json: Permission denied
```

### Expected behavior

Start, like versions prior to 4.0.0 did with the same config.

### How to reproduce

```
docker run --rm -it -e SHLINK_SERVER_URL="https://go.example.com" -e SHLINK_SERVER_API_KEY=foo shlinkio/shlink-web-client:4.0.0
```

### PR DESCRIPTION
**Pull request by jinnatar**
Fixes #1046

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -9,7 +9,13 @@ ARG UID=101
 LABEL maintainer="Alejandro Celaya <alejandro@alejandrocelaya.com>"
 USER root
 RUN rm -r /usr/share/nginx/html && rm /etc/nginx/conf.d/default.conf
-USER $UID
 COPY config/docker/nginx.conf /etc/nginx/conf.d/default.conf
 COPY scripts/docker/servers_from_env.sh /docker-entrypoint.d/30-shlink-servers-json.sh
 COPY --from=node /shlink-web-client/build /usr/share/nginx/html
+
+# Allow the nginx user to create new files in the html dir, but not modify existing files.
+# This is required by 30-shlink-servers-json.sh
```
Comment (acelaya) [inline Dockerfile]: I think this comment is misleading, as the user can still modify files directly in the `html` folder, but not its subfolders.

**Review by acelaya — CHANGES_REQUESTED** [review]
I have tested this, and it serves the purpose of not failing when env vars are provided, and allowing the servers.json file to be created.

However, by setting the permissions this way, the 101 user can still modify/delete any file directly placed under `/usr/share/nginx/html`, which defeats a bit the purpose of running as non-root (but not completely).

I'm thinking perhaps the `servers.json` file could be created with an empty array on it, set write permissions only to that specific file instead of the whole folder, and refactor `servers_from_env.sh` to overwrite the contents of the file (if not already doing it).

This may have some side effects though, as it should be possible to also mount this file as a volume, so this scenario should be tested.

File: Dockerfile (hunk 1 above)
Comment (jinnatar) [reply Dockerfile]: Correct, seems I was a bit too tired at the time and wrongly assumed dir level write would not confer edit, but indeed it does. Switched to creating an empty server list by default.

**Comment by jinnatar** [comment]
Switched to the suggested approach which indeed is more secure. `servers_from_env.sh` already overwrites the contents, so no changes needed there.

If the file is volume mounted there's nothing we can do really. As long as it's readable by `101`, it works fine. The only dodgy part is if the file is volume mounted, has no write access and env variables are also given, then it will error out due to the entrypoint attempting to overwrite it. But that's something no one should run into and if they do, they deserve it. :-)

**Comment by acelaya** [comment]
This works great! Thanks a lot for the contribution @jinnatar 🙂 

I'll wait for a couple of days just in case some other bug is reported, and then release this with v4.0.1

**Pull request merged**